app/helpers/turbo/streams/action_helper.rb

```diff
@@ -6,11 +6,20 @@ module Turbo::Streams::ActionHelper
   #
   #   turbo_stream_action_tag "replace", target: "message_1", template: %(<div id="message_1">Hello!</div>)
   #   # => <turbo-stream action="replace" target="message_1"><template><div id="message_1">Hello!</div></template></turbo-stream>
-  def turbo_stream_action_tag(action, target:, template: nil)
+  #
+  # If remove_if_present is specified, the helper will generate 2 tags (remove and the original action)
+  #
+  def turbo_stream_action_tag(action, target:, template: nil, remove_if_present: nil)
     target   = convert_to_turbo_stream_dom_id(target)
     template = action.to_sym == :remove ? "" : "<template>#{template}</template>"
 
-    %(<turbo-stream action="#{action}" target="#{target}">#{template}</turbo-stream>).html_safe
+    if remove_if_present
```

Comment:
As an alternative to accepting and passing around an `remove_if_present:` option, would it be safe to infer the value from `action.in? %[ append prepend ]` within the `ActionHelper#turbo_stream_action_tag` method?

That way, callers are not responsible for making the decision themselves, and `prepend` and `append` actions are always going to remove counterparts with matching `[id]` attributes.

Reply:
Right... the issue is how to recover the dom id of the rendered element itself (the `target` parameter  is the id of the container, not the element).
If it's not passed through the call chain, I'm not sure how to recover it. Are you thinking of parsing the template to extract the id of the element ?

Reply:
Ah, that's a great point. Maybe the logic belongs higher in the call stack, closer to the code that maps the ActiveModel instance to the broadcast operation.